### PR TITLE
Add ravel_multi_index

### LINKED ISSUE #2557: Adding ravel_multi_index and unravel_index
Would be nice to have Dask implementations of [ravel_multi_index]( https://docs.scipy.org/doc/numpy/reference/generated/numpy.ravel_multi_index.html ) and [unravel_index]( https://docs.scipy.org/doc/numpy/reference/generated/numpy.unravel_index.html ). These are particularly nice to have as we are starting to look at Dask index arrays ( e.g. https://github.com/dask/dask/pull/2539 ).

### PR DESCRIPTION
**Pull request by Illviljan**
Closes #2557.
- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

I've seen a couple of ways implementing this functionality when trying to understand what's going on:
* https://github.com/pydata/sparse/blob/9dc40e15a04eda8d8efff35dfc08950b4c07a810/sparse/_coo/common.py#L56-L64
* https://github.com/xgcm/xhistogram/blob/7ba35baf0d40b8b9d9d5fa22ffe48b6fb11bc7d3/xhistogram/duck_array_ops.py#L39

I don't know if these are better alternatives. I mainly just went with getting the numpy version working, and I think it would start to get messy if we want to support all the options that numpy has without using `np.ravel_multi_index`.

### CODE REVIEW

File: dask/array/tests/test_routines.py [hunk 1]
```diff
@@ -1503,6 +1503,32 @@ def test_unravel_index():
         assert_eq(darr.vindex[d_indices], arr[indices])
 
 
+def test_ravel_multi_index():
+    for arr, chunks, kwargs in [
```
Comment (jsignell) [inline dask/array/tests/test_routines.py:1507]: Can you make this a parameterized test instead?

**Review by jsignell — COMMENTED** [review]
Thanks for opening this PR! I think the functionality will be very nice. I just have a comment about the test.

File: dask/array/tests/test_routines.py (hunk 1 above)
Comment (Illviljan) [reply dask/array/tests/test_routines.py:1507]: In a pytest way? I'm not used to pytest yet, is there a good example of what you mean?

Comment (jsignell) [reply dask/array/tests/test_routines.py]: Essentially you just move the loop outside of the test. https://github.com/dask/dask/blob/3a08bb32e736d51a758281a3abc0fc0dbec04efc/dask/array/tests/test_routines.py#L226-L256 for example

Comment (Illviljan) [reply dask/array/tests/test_routines.py]: Should be fixed now.

**Comment by jsignell** [comment]
Thanks for this PR @Illviljan!

**Pull request merged**